dispatcher.go

```diff
@@ -214,7 +267,7 @@ func (md *multicastDispatcher[T]) CloseEventStreams() {
 
 // newMulticastDispatcher creates a new Dispatcher[T]
 func newMulticastDispatcher[T any]() Dispatcher[T] {
-	in := make(chan *dispatchedEvent[T], 5)
+	in := make(chan *dispatchedEvent[T], 20)
```

Comment:
Increasing this, while dispatching 5 events of the same type at once seemed semi-aggressive, I'd rather provide more buffer room here. I want to make sure we maintain a good dispatch to receiver ratio. I will formalize a benchmark at some point.